src/rules/migrate_root_attributes.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @title Move user_metadata attributes to profile root attributes.
+ * @overview Moves select data from user_metadata to profile root attributes (family_name, given_name, name, nickname and picture).
+ * @gallery true
+ * @category enrich profile
+ *
+ * This rule moves select data from user_metadata to profile root attributes (family_name, given_name, name, nickname and picture).
+ * Verify the field mapping before enabling this rule.
+ * The rule will determine if there is a mapped field on the user_metadata before the update.
+ * Important:
+ *
+ * 1- The rule updates the profile root attribute with the mapped field from user_metadata.
+ * 2- The mapped fields from user_metadata will be removed following the update.
+ * 3- This rule will be executed on each login event. Consider using it only if you use a custom signup form or Authentication Signup API.
+ */
+function (user, context, cb) {
+  // Field Mapping, the property is the root attribute and the value is the field name on user_metadata.
+  // You can change the value in case you don't have the same name on user_metadata.
+  const fieldMapping = {
+    family_name: 'family_name',
+    given_name: 'given_name',
+    name: 'name',
+    nickname: 'nickname',
+    picture: 'picture'
+  };
+
+  if (needMigration(user)) {
+    const ManagementClient = require('auth0@2.9.1').ManagementClient;
+    const management = new ManagementClient({
+      domain: auth0.domain,
+      token: auth0.accessToken
+    });
+
+    management.updateUser(
+      { id: user.user_id }, generateUserPayload(user), function (err, user) {
+        if ( err ) {
+          cb(err);
+        } else {
+          cb(null, user, context);
```

Comment:
did you check if it's ok to return a new `user` object (api2's result) instead of the original one?

Reply:
Yes, I did, it returns the updated user.

Reply:
I changed the code to use the original one.